calibrate.py

```diff
@@ -0,0 +1,438 @@
+                'mod': 0, 'scancode': 15, 'key': pygame.K_TAB, 'unicode': "u'\t'"}))
+            pygame.event.post(pygame.event.Event(pygame.KEYDOWN,{
+                'mod': 0, 'scancode': 15, 'key': pygame.K_TAB, 'unicode': "u'\t'"}))
+        if button['dup']:
+            pygame.event.post(pygame.event.Event(pygame.KEYDOWN,{
+                'mod': 1, 'scancode': 15, 'key': pygame.K_TAB, 'unicode': "u'\t'"}))
+            pygame.event.post(pygame.event.Event(pygame.KEYDOWN,{
+                'mod': 1, 'scancode': 15, 'key': pygame.K_TAB, 'unicode': "u'\t'"}))
+
+    def update_value(self):
+        for ctrl in self.controls:
+            i = ctrl['index']
+            self.processor.colour_value[i % 2][int(i/2)] = ctrl['ctrl'].value
+            self.processor.colour_value[i % 2][int(i/2)] = int(self.processor.colour_value[i % 2][int(i/2)])
+
+    def save_values(self):
+        if self.colour_index is not None:
+            colour = self.colour_radio_buttons[self.colour_index]['label']
+            self.colour_values[colour] = self.processor.colour_value
+            data = self.colour_values
+            filename = self.file_radio_buttons[self.file_index]['label'] + ".json"
+            with open(filename, 'w') as f:
+                json.dump(data, f)
+            save_label = sgc.Label(text="saved", pos=(15, 160), col=(255,255,255))
+            save_label.add(101)
+            self.clock = pygame.time.Clock()
+            timer = self.clock.tick(30)
+            sgc.update(timer)
+            time.sleep(1)
+            save_label.remove(fade=False) 
+
+    def update_display(self):
+        for radio in self.file_radio_buttons:
+            if radio['btn'].has_focus():
+                self.display_values()
+                self.logger.info("%s file selected for colour editing" % radio['label'])
+                if self.colour_label:
+                    self.colour_label.remove(fade=False)
+        for radio in self.colour_radio_buttons:
+            if radio['btn'].has_focus():
+                self.processor.colour_value = self.colour_values[radio['label']]
+                label_text = str(self.colour_values[radio['label']])
+                if self.colour_label and self.colour_label.active():
+                    self.colour_label.text = label_text
+                else:
+                    self.colour_label = sgc.Label(text=label_text, pos=(15, 150), col=(255,255,255))
+                    self.colour_label.add(100)
+                self.logger.info("%s value selected for editing" % radio['label'])
+
+    def display_files(self, index=None):
+        file_path = os.path.dirname(os.path.realpath(__file__))
+        all_files = os.listdir(file_path)
+        self.file_radio_buttons = []
+        button_index = 0
+        for filename in os.listdir(file_path):
+            if filename.endswith('.json') and filename <> 'calibration.json': 
+                display_name = filename[:len(filename)-5] #trim filetype off
```

Comment:
don't do this by hand. it's brittle and error prone. Python has your back here:
https://docs.python.org/2/library/os.path.html#os.path.split